internal/pkg/cli/job_init.go

```diff
@@ -0,0 +1,297 @@
+		dockerfiles,
+		prompt.WithFinalMessage("Dockerfile:"),
+	)
+	if err != nil {
+		return fmt.Errorf("select Dockerfile: %w", err)
+	}
+
+	o.DockerfilePath = sel
+
+	return nil
+}
+
+// RecommendedActions returns follow-up actions the user can take after successfully executing the command.
+func (o *initJobOpts) RecommendedActions() []string {
+	return []string{
+		fmt.Sprintf("Update your manifest %s to change the defaults.", color.HighlightResource(o.manifestPath)),
+		fmt.Sprintf("Run %s to deploy your job to a %s environment.",
+			color.HighlightCode(fmt.Sprintf("copilot job deploy --name %s --env %s", o.Name, defaultEnvironmentName)),
+			defaultEnvironmentName),
+	}
+}
+
+// BuildJobInitCmd builds the command for creating a new service.
+func BuildJobInitCmd() *cobra.Command {
+	vars := initJobVars{
+		GlobalOpts: NewGlobalOpts(),
+	}
+	cmd := &cobra.Command{
+		Use:   "init",
+		Short: "Creates a new scheduled job in an application.",
+		Example: `
+  Create a "reaper" scheduled task to run once per day.
+  /code $ copilot job init --name reaper --dockerfile ./frontend/Dockerfile --rate "1 day"
+
+  Create a "report-generator" scheduled task with retries.
+  /code $ copilot job init --name report-generator --rate "1 day" --retries 3 --timeout `,
+		RunE: runCmdE(func(cmd *cobra.Command, args []string) error {
+			opts, err := newInitJobOpts(vars)
+			if err != nil {
+				return err
+			}
+			if err := opts.Validate(); err != nil { // validate flags
+				return err
+			}
+			if err := opts.Ask(); err != nil {
+				return err
+			}
+			if err := opts.Execute(); err != nil {
+				return err
+			}
+			log.Infoln("Recommended follow-up actions:")
+			for _, followup := range opts.RecommendedActions() {
+				log.Infof("- %s\n", followup)
+			}
+			return nil
+		}),
+	}
+	cmd.Flags().StringVarP(&vars.Name, nameFlag, nameFlagShort, "", jobFlagDescription)
+	cmd.Flags().StringVarP(&vars.DockerfilePath, dockerFileFlag, dockerFileFlagShort, "", dockerFileFlagDescription)
+
+	requiredFlags := pflag.NewFlagSet("Required Flags", pflag.ContinueOnError)
+	requiredFlags.AddFlag(cmd.Flags().Lookup(nameFlag))
+	requiredFlags.AddFlag(cmd.Flags().Lookup(dockerFileFlag))
+
+	cmd.Annotations = map[string]string{
+		// The order of the sections we want to display.
+		"sections": fmt.Sprintf(`Required,%s`, strings.Join(manifest.ServiceTypes, ",")),
+		"Required": requiredFlags.FlagUsages(),
+	}
+	cmd.SetUsageTemplate(`{{h1 "Usage"}}{{if .Runnable}}
+  {{.UseLine}}{{end}}{{$annotations := .Annotations}}{{$sections := split .Annotations.sections ","}}{{if gt (len $sections) 0}}
+
+{{range $i, $sectionName := $sections}}{{h1 (print $sectionName " Flags")}}
+{{(index $annotations $sectionName) | trimTrailingWhitespaces}}{{if ne (inc $i) (len $sections)}}
+
+{{end}}{{end}}{{end}}{{if .HasAvailableInheritedFlags}}
+
+{{h1 "Global Flags"}}
+{{.InheritedFlags.FlagUsages | trimTrailingWhitespaces}}{{end}}{{if .HasExample}}
+
+{{h1 "Examples"}}{{code .Example}}{{end}}
+`)
```

Comment:
I don't think we need this section, it was only introduced so that we have a fancy help menu for our different service types.

For job init, we should be able to use cobra's default command template.